infrastructure/src/main/java/org/corfudb/infrastructure/configuration/ServerConfiguration.java

```diff
@@ -0,0 +1,722 @@
+        conf.setTruststore(configProperties.getProperty(TRUSTSTORE));
+        conf.setTruststorePasswordFile(configProperties.getProperty(TRUSTSTORE_PASSWORD_FILE));
+        conf.setEnableTlsMutualAuth(Boolean.parseBoolean(configProperties.getProperty(ENABLE_TLS_MUTUAL_AUTH, "false")));
+        conf.setEnableSaslPlainTextAuth(Boolean.parseBoolean(configProperties.getProperty(ENABLE_SASL_PLAIN_TEXT_AUTH, "false")));
+        conf.setSaslPlainTextUserFile(configProperties.getProperty(SASL_PLAIN_TEXT_USERNAME_FILE));
+        conf.setSaslPlainTextPasswordFile(configProperties.getProperty(SASL_PLAIN_TEXT_PASSWORD_FILE));
+        conf.setSequencerCacheSize(Integer.parseInt(configProperties.getProperty(SEQUENCER_CACHE_SIZE,DEFAULT_SEQUENCER_CACHE_SIZE)));
+
+        conf.setStateTransferBatchSize(Integer.parseInt(configProperties.getProperty(STATE_TRANSFER_BATCH_SIZE, DEFAULT_STATE_TRANSFER_BATCH_SIZE)));
+
+        String implementationType = configProperties.getProperty(CHANNEL_IMPLEMENTATION,DEFAULT_CHANNEL_IMPLEMENTATION_TYPE);
+        conf.setChannelImplementation(ChannelImplementation.valueOf(implementationType.toUpperCase()));
+
+        conf.setHandshakeTimeout(Integer.parseInt(configProperties.getProperty(HANDSHAKE_TIMEOUT, DEFAULT_HANDSHAKE_TIMEOUT)));
+        conf.setClusterId(configProperties.getProperty(CLUSTER_ID, DEFAULT_CLUSTER_ID));
+        conf.setTlsCiphers(configProperties.getProperty(TLS_CIPHERS,DEFAULT_TLS_CIPHERS));
+        conf.setTlsProtocols(configProperties.getProperty(TLS_PROTOCOLS, DEFAULT_TLS_PROTOCOLS));
+
+        conf.setEnableMetrics(Boolean.parseBoolean(configProperties.getProperty(ENABLE_METRICS, "false")));
+        conf.setSnapshotBatchSize(Integer.parseInt(configProperties.getProperty(SNAPSHOT_BATCH_SIZE, Integer.toString(DEFAULT_MAX_NUM_MSG_PER_BATCH))));
+        conf.setLockLeaseDuration(Integer.parseInt(configProperties.getProperty(LOCK_LEASE_DURATION, Integer.toString(Lock.leaseDuration))));
+        conf.setThreadPrefix(configProperties.getProperty(THREAD_PREFIX, DEFAULT_THREAD_PREFIX));
+        conf.setMetadataRetention(Integer.parseInt(configProperties.getProperty(METADATA_RETENTION, DEFAULT_METADATA_RETENTION)));
+
+        conf.setServerPort(Integer.parseInt(configProperties.getProperty(SERVER_PORT)));
+
+        return conf;
+    }
+
+    public static ServerConfiguration getServerConfigFromMap(Map<String, Object> opts) {
+        Properties configProperties = new Properties();
+        //in the future, if more formats need to be supported, this mapping can be passed as a parameter
+        Map<String, String> optionsToPropertiesMapping = getOptionsToPropertiesMapping();
+        Set<String> flags = ImmutableSet.of("--memory", "--single", "--enable-tls",
+                "--enable-tls-mutual-auth","--enable-sasl-plain-text-auth", "--metrics");
+        Set<String> inverseFlags = ImmutableSet.of("--no-verify", "--no-sync", "--no-auto-commit");
+        for (String key : opts.keySet()) {
```

Comment:
Since you are utilizing both key and value in this loop, why not iterate over the key-value-pair set instead of just the key set?